extensions/blocks/recurring-payments/index.js

```diff
@@ -83,3 +88,73 @@ export const SUPPORTED_CURRENCY_LIST = [
 	'SEK',
 	'SGD',
 ];
+
+/**
+ * Stripe's minimum transaction amounts per currency in the smallest unit for
+ * that currency.
+ *
+ * https://stripe.com/docs/currencies#minimum-and-maximum-charge-amounts
+ *
+ * @type { [currenty: string]: number }
+ */
+export const MINIMUM_TRANSACTION_PER_CURRENCY = {
+	USD: 0.5,
+	AUD: 0.5,
+	BRL: 0.5,
+	CAD: 0.5,
+	CHF: 0.5,
+	DKK: 2.5,
+	EUR: 0.5,
+	GBP: 0.3,
+	HKD: 4.0,
+	INR: 0.5,
+	JPY: 50,
+	MXN: 10,
+	MYR: 2,
+	NOK: 3.0,
+	NZD: 0.5,
+	PLN: 2.0,
+	SEK: 3.0,
+	SGD: 0.5,
+};
+
+/**
+ * Compute a list of currency value and display labels.
+ *
+ * - `value` is the currency's three character code
+ * - `label` is the user facing representation.
+ *
+ * @typedef {{value: string, label: string}} CurrencyDetails
+ *
+ * @type Array<CurrencyDetails>
+ */
+export const CURRENCY_OPTIONS = SUPPORTED_CURRENCY_LIST.map( value => {
+	const { symbol } = getCurrencyDefaults( value );
+	// if symbol is equal to the code (e.g., 'CHF' === 'CHF'), don't duplicate it.
+	// trim the dot at the end, e.g., 'kr.' becomes 'kr'
+	const label = symbol === value ? value : `${ value } ${ trimEnd( symbol, '.' ) }`;
+	return { value, label };
+} );
+
+/**
+ * Returns the minimum transaction amount for the given currency. If currency is not one of the
+ * known types it returns ...
```

Comment:
@artpi Should we default to a value if the currency is not one of the known options? This shouldn't happen in practice.

Reply:
The currency list gets populated from these constants above. So that should not be happening and we don't want want fallbacks - we want errors in this case :)